src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -897,6 +897,31 @@ static int ensure_sane_request(sd_dhcp_server *server, DHCPRequest *req, DHCPMes
         return 0;
 }
 
+static void request_set_timestamp(DHCPRequest *req, const triple_timestamp *timestamp) {
+        assert(req);
+
+        if (timestamp && triple_timestamp_is_set(timestamp))
+                req->timestamp = *timestamp;
+        else
+                triple_timestamp_get(&req->timestamp);
```

Comment:
I think the naming of the `*_timestamp_get()` functions is confusing. But that's a separate topic. It's called correctly here.

Reply:
I prepared a patch to change this. I'll submit it once this PR is merged.